ssg/build_yaml.py

```diff
@@ -1085,14 +1085,17 @@ def __init__(self, id_):
     def from_yaml(cls, yaml_file, env_yaml=None):
         yaml_file = os.path.normpath(yaml_file)
 
-        yaml_contents = open_and_macro_expand(yaml_file, env_yaml)
-        if yaml_contents is None:
-            return None
-
         rule_id, ext = os.path.splitext(os.path.basename(yaml_file))
         if rule_id == "rule" and ext == ".yml":
             rule_id = get_rule_dir_id(yaml_file)
 
+        if env_yaml:
+            env_yaml["rule_id"] = rule_id
```

Comment:
In most places, prior to modification (such as https://github.com/ComplianceAsCode/content/blob/master/ssg/build_ovals.py#L269 or https://github.com/ComplianceAsCode/content/blob/master/ssg/build_remediations.py#L221) we'd duplicate `env_yaml` and call it `local_env_yaml`. This keeps a constant dict from the caller's perspective (right now, you're modifying the caller's copy as well).

I'd perhaps suggest adding a `self.local_env_yaml = None` member and if `env_yaml` is specified, duplicate it into the local one. Places that load a rule can then use a single consistent env_yaml (populated from the `Rule` class) in most places.

My 2c.

Reply:
That said, I am definitely in favor of this! Thanks @ggbecker!

Reply:
what do you think about 0aa0087c01528a34364f28e3a419000772195351 ?

Reply:
Looks great, thanks!